applications/StructuralMechanicsApplication/tests/cpp_tests/test_constitutive_law_utilities.cpp

```diff
@@ -71,5 +71,44 @@ KRATOS_TEST_CASE_IN_SUITE(ConstitutiveLawUtilitiesPrincipalStresses, KratosStruc
     KRATOS_CHECK_LESS_EQUAL((-9.00151e+06 - principal_stress_vector[1])/principal_stress_vector[1], tolerance);
     KRATOS_CHECK_LESS_EQUAL((-2.53163e+07 - principal_stress_vector[2])/principal_stress_vector[2], tolerance);
 }
+
+/**
+* Check the correct calculation of the CL utilities Hencky and Biot strains
+*/
+KRATOS_TEST_CASE_IN_SUITE(ConstitutiveLawUtilitiesHenckyAndBiot, KratosStructuralMechanicsFastSuite)
+{
+    // Declaration of Cauchy tensor and strain vector
+    Matrix C = ZeroMatrix(3, 3);
+    Vector strain_vector(6);
+
+    // Fill Cauchy tensor
+    C(0,0) = 1.1;
+    C(1,1) = 1.0;
+    C(2,2) = 1.0;
+    C(1,0) = -1.0e-1;
+    C(0,1) = -1.0e-1;
+
+    // Compute Hencky
+    ConstitutiveLawUtilities<6>::CalculateHenckyStrain(C, strain_vector);
+
+//     // Debug
+//     KRATOS_WATCH(strain_vector)
+
+    KRATOS_CHECK_LESS_EQUAL((0.0454445 - strain_vector[0])/strain_vector[0], tolerance);
+    KRATOS_CHECK_LESS_EQUAL((-0.0023557 - strain_vector[1])/strain_vector[1], tolerance);
+    KRATOS_CHECK_LESS_EQUAL((0.0 - strain_vector[2])/strain_vector[2], tolerance);
+    KRATOS_CHECK_LESS_EQUAL((-0.0956005 - strain_vector[3])/strain_vector[3], tolerance);
+
+    // Compute Biot
+    ConstitutiveLawUtilities<6>::CalculateBiotStrain(C, strain_vector);
+
+//     // Debug
+//     KRATOS_WATCH(strain_vector)
+
+    KRATOS_CHECK_LESS_EQUAL((0.0476699 - strain_vector[0])/strain_vector[0], tolerance);
+    KRATOS_CHECK_LESS_EQUAL((-0.00119458 - strain_vector[1])/strain_vector[1], tolerance);
+    KRATOS_CHECK_LESS_EQUAL((0.0 - strain_vector[2])/strain_vector[2], tolerance);
```

Comment:
this check is error prone due to zero division

Reply:
True